File: Scripts/newTask/regular-expressions/perl/regular-expressions.pl
$string = "this is a matching string";

for( $a=0; $a < 10000000; $a = $a + 1) {
if ($string =~ /string$/) {
}

$string =~ s/ string/ istyfied/; # makes "I am a string" into "I am another string"
}

File: Scripts/newTask/regular-expressions/go/regular-expressions.go
package main
import "regexp"

func main() {
  str := "this is a matching string"

  // Test
  for i := 0; i < 1000000; i++ {
  matched, _ := regexp.MatchString(".*string$", str)
  if matched { }

  // Substitute
  pattern := regexp.MustCompile("matching")
  result := pattern.ReplaceAllString(str, "istyfied")
  _ = result
  }

}
